app/models/enrollment/dgfip.rb

```diff
@@ -21,7 +21,7 @@ def fields_validation
 
         errors[:siren] << "Vous devez renseigner le SIREN de votre organisation avant de continuer" unless siren.present?
         errors[:demarche] << "Vous devez renseigner la description de la démarche avant de continuer" unless demarche && demarche['description'].present?
-        errors[:demarche] << "Vous devez renseigner le fondement juridique de la démarche avant de continuer" unless demarche && demarche['fondement_juridique'].present?
+        errors[:demarche] << "Vous devez renseigner le fondement juridique de la démarche avant de continuer" unless (demarche && demarche['fondement_juridique'].present?) || documents.where(type: 'Document::LegalBasis').present?
```

Comment:
can you merge app/models/enrollment/api_particulier.rb with app/models/enrollment/dgfip.rb ?